binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/bootstrap/KafkaStreamsBinderBootstrapTest.java

```diff
@@ -192,11 +192,18 @@ public Consumer<KTable<Map<String, String>, Map<String, String>>> input2() {
 			};
 		}
 
+	}
+
+	// Testing the scenario reported by https://github.com/spring-cloud/spring-cloud-stream/issues/2737
```

Comment:
Does this test modification cover both of the areas that have the new treatment?

Reply:
Ya, it does. I wanted to make sure that by moving the function out into a super class, it is detected by both code paths.